packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-trigger-template.ts

```diff
@@ -0,0 +1,163 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { $TSContext, $TSObject, FeatureFlags, JSONUtilities, pathManager } from 'amplify-cli-core';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+import { prepareApp } from '@aws-cdk/core/lib/private/prepare-app';
+import { AuthTriggerConnection } from '../service-walkthrough-types';
+import { CustomResource, CustomResourceProps, RemovalPolicy } from '@aws-cdk/core';
+import * as logs from '@aws-cdk/aws-logs';
+import { Permission } from '@aws-cdk/aws-lambda';
+
+type CustomResourceAuthStackProps = Readonly<{
+  description: string;
+  authLambdaConfig: AuthTriggerConnection[];
+}>;
+
+const CFN_TEMPLATE_FORMAT_VERSION = '2010-09-09';
+
+export class CustomResourceAuthStack extends cdk.Stack {
+  constructor(scope: cdk.Construct, id: string, props: CustomResourceAuthStackProps) {
+    super(scope, id, props);
+    this.templateOptions.templateFormatVersion = CFN_TEMPLATE_FORMAT_VERSION;
+
+    const env = new cdk.CfnParameter(this, 'env', {
+      type: 'String',
+    });
+
+    const userpoolId = new cdk.CfnParameter(this, 'userpoolId', {
+      type: 'String',
+    });
+
+    const userpoolArn = new cdk.CfnParameter(this, 'userpoolArn', {
+      type: 'String',
+    });
+
+    new cdk.CfnCondition(this, 'ShouldNotCreateEnvResources', {
+      expression: cdk.Fn.conditionEquals(env, 'NONE'),
+    });
+
+    props.authLambdaConfig.forEach(config => {
+      const fnName = new cdk.CfnParameter(this, `function${config.fnName}Name`, {
+        type: 'String',
+      });
+      const fnArn = new cdk.CfnParameter(this, `function${config.fnName}Arn`, {
+        type: 'String',
+      });
+      createPermissionToInvokeLambda(this, fnName, userpoolArn, config);
+      config.fnArn = fnArn.valueAsString;
+    });
+
+    createCustomResource(this, props.authLambdaConfig, userpoolId);
+  }
+
+  toCloudFormation() {
+    prepareApp(this);
+    return this._toCloudFormation();
+  }
+}
+
+export async function generateNestedAuthTriggerTemplate(context: $TSContext, category: string, request: $TSObject) {
+  const cfnFileName = 'auth-trigger-cloudformation-template.json';
+  const targetDir = path.join(pathManager.getBackendDirPath(), category, request.resourceName);
+  const authTriggerCfnFilePath = path.join(targetDir, cfnFileName);
+  const { authLambdaConfig } = request;
+  if (authLambdaConfig !== undefined && authLambdaConfig.length !== 0) {
+    const cfnObject = await createCustomResourceforAuthTrigger(context, JSON.parse(authLambdaConfig));
+    JSONUtilities.writeJson(authTriggerCfnFilePath, cfnObject);
+  } else {
+    // delete the custom stack template if the triggers arent defined
+    if (fs.existsSync(authTriggerCfnFilePath)) {
+      fs.unlinkSync(authTriggerCfnFilePath);
+    }
```

Comment:
Can you call `unlinkSync()` unconditionally and handle/ignore the error if the file doesn't exist. This pattern adds an extra file system access, and introduces potential race conditions.